lib/gds_api/json_client.rb

```diff
@@ -246,7 +246,11 @@ def do_request(method, url, params = nil, additional_headers = {})
       when :get, :delete
         default_headers = self.class.default_request_headers
       else
-        default_headers = self.class.default_request_with_json_body_headers
+        if params.respond_to?(:has_key?) && params[:multipart] == true
```

Comment:
Minor: It seems slightly odd to check for the existence of the `has_key?` method. Presumably you can't use `params.respond_to?(:[])` here because a `String` also responds to that method...? If so, I think it might make more sense to use `params.is_a?(Hash)` in this case.

Reply:
It looks like this code is deleted in the next commit, so you can probably ignore my previous comment.